### PR TITLE
Upgrade Mattermost to v8.1 ESR

### PR DESCRIPTION
**Pull request by ArtofBugs**

### CODE REVIEW

**Comment by ArtofBugs** [comment]
> In v8.0, the following repositories are merged into one: mattermost-server, mattermost-webapp and mmctl. Developers should read the updated [Developer Guide](https://developers.mattermost.com/contribute/developer-setup/) for details.

Looks like a separate mmctl isn't supported anymore... for future reference, the migration guide is here https://developers.mattermost.com/contribute/monorepo-migration-notes/

File: resources/mattermost.rb [hunk 1]
```diff
@@ -41,18 +40,6 @@
...
-    prefix_root '/opt'
-    prefix_home '/opt'
-    strip_components 0
-    version new_resource.mmctl_version
-  end
-
-  link '/usr/local/bin/mmctl' do
-    to '/opt/mmctl/mmctl'
-  end
-
```
Comment (ramereth) [inline resources/mattermost.rb]: So this actually got moved to the single tarball. If you download https://releases.mattermost.com/8.1.7/mattermost-8.1.7-linux-amd64.tar.gz and then extract the file at `mattermost/bin/mmctl` that will get you what you need. I believe the `ark` resource allows you to grab a specific binary. Can you please bring this back? We'll just set the mmctl version to the mattermost semver version.

**Review by ramereth — CHANGES_REQUESTED** [review]

File: resources/mattermost.rb [hunk 2]
```diff
@@ -42,7 +42,10 @@
   package %w(rsync tar)
 
   ark 'mmctl' do
-    url "https://github.com/mattermost/mmctl/releases/download/v#{new_resource.mmctl_version}/linux_amd64.tar"
+    action :cherry_pick
+    path '/opt/mmctl/'
+    creates 'mattermost/bin/mmctl'
+    url "https://releases.mattermost.com/#{new_resource.mmctl_version}/mattermost-#{new_resource.mmctl_version}-linux-amd64.tar.gz"
```
Comment (ramereth) [inline resources/mattermost.rb:45]: Can you please move this to the top like it was before?

**Comment by ramereth** [comment]
@ArtofBugs you need to update the link resource to point to the new location for mmctl

**Comment by ArtofBugs** [comment]
> @ArtofBugs you need to update the link resource to point to the new location for mmctl

Yep, finally figured out how the installation path actually works... 😅

**Review by ramereth — APPROVED** [review]
I will coordinate with customer and merge/deploy this

**Comment by ramereth** [comment]
!bump minor ~

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `main` and has automatically performed a minor-level version bump to v1.1.0.  Have a nice day!

**Pull request merged**